diff --git a/nvim/my-colors/colors/nord.vim b/nvim/my-colors/colors/nord.vim
index c3ab369..003400a 100644
--- a/nvim/my-colors/colors/nord.vim
+++ b/nvim/my-colors/colors/nord.vim
@@ -17,6 +17,181 @@ let g:colors_name = "nord"
 let s:nord_vim_version="0.6.0"
 set background=dark
 
+function! s:hl(group, ctermfg, ctermbg, attr)
+  if a:ctermfg != ""
+    exec "hi " . a:group . " ctermfg=" . a:ctermfg
+  endif
+  if a:ctermbg != ""
+    exec "hi " . a:group . " ctermbg=" . a:ctermbg
+  endif
+  if a:attr != ""
+    exec "hi " . a:group . " gui=" . a:attr . " cterm=" . a:attr
+  endif
+endfunction
+
+" TODO: remove
+let s:nord1_term = "0"
+let s:nord3_term = "8"
+let s:nord5_term = "7"
+let s:nord6_term = "15"
+let s:nord7_term = "14"
+let s:nord8_term = "6"
+let s:nord9_term = "4"
+let s:nord10_term = "12"
+let s:nord11_term = "1"
+let s:nord12_term = "11"
+let s:nord13_term = "3"
+let s:nord14_term = "2"
+let s:nord15_term = "5"
+
+let s:color_bg          = "NONE"
+let s:color_bg_alt      = "0"
+let s:color_bg_accent   = "5"
+
+let s:color_primary     = "7"
+let s:color_secondary   = "8"
+let s:color_accent      = "9"
+
+let s:color_success     = "2"
+let s:color_warn        = "11"
+let s:color_error       = "9"
+
+let s:color_syntax_a    = "7"
+let s:color_syntax_b    = "4"
+let s:color_syntax_c    = "1"
+let s:color_syntax_d    = "102"
+
+" Attributes
+call s:hl("Bold", "", "", "bold")
+call s:hl("Italic", "", "", "italic")
+call s:hl("Underline", "", "", "underline")
+
+"+--- Editor ---+
+call s:hl("Normal", "NONE", s:color_bg, "")
+call s:hl("LineNr", s:color_secondary, "NONE", "")
+call s:hl("ColorColumn", "", s:color_bg_alt, "")
+call s:hl("Cursor", "", "NONE", "")
+call s:hl("CursorLine", "", s:color_bg_alt, "NONE")
+call s:hl("iCursor", "", "NONE", "")
+call s:hl("MatchParen", s:color_primary, s:color_bg_accent, "")
+call s:hl("NonText", s:color_secondary, "", "")
+call s:hl("SpecialKey", s:color_secondary, "", "")
+call s:hl("PMenu", "NONE", s:color_bg_alt, "")
+call s:hl("PmenuSbar", "NONE", s:color_bg_alt, "")
+call s:hl("PmenuThumb", "NONE", s:color_secondary, "")
+call s:hl("PMenuSel", s:color_bg_alt, s:color_bg_accent, "")
+
+
+call s:hl("Spell", "", "NONE", "undercurl")
+hi! link SpellBad Spell
+hi! link SpellLocal Spell
+hi! link SpellRare Spell
+
+call s:hl("Visual", "", s:color_bg_alt, "")
+call s:hl("VisualNOS", "", s:color_bg_alt, "")
+
+"+- Neovim Support -+
+call s:hl("healthError", s:color_error, "NONE", "")
+call s:hl("healthSuccess", s:color_success, "NONE", "")
+call s:hl("healthWarning", s:color_warn, "NONE" , "")
+
+"+--- Gutter ---+
+call s:hl("CursorColumn", "NONE", s:color_bg_alt, "")
+call s:hl("CursorLineNr", s:color_primary, s:color_bg_alt, "")
+call s:hl("Folded", s:color_secondary, s:color_bg_alt, "bold")
+call s:hl("FoldColumn", s:color_secondary, "NONE", "")
+call s:hl("SignColumn", s:color_bg_alt, "NONE", "")
+
+"+--- Navigation ---+
+call s:hl("Directory", s:color_primary, "NONE", "")
+
+"+--- Prompt/Status ---+
+call s:hl("EndOfBuffer", s:color_secondary, "NONE", "")
+call s:hl("ErrorMsg", "NONE", s:color_error, "")
+call s:hl("ModeMsg", "", "", "")
+call s:hl("MoreMsg", "", "", "")
+call s:hl("Question", "NONE", "", "")
+call s:hl("StatusLine", s:color_bg_alt, s:color_bg_accent, "NONE")
+call s:hl("StatusLineNC", "NONE", s:color_secondary, "")
+call s:hl("WarningMsg", s:color_bg_alt, s:color_warn, "")
+call s:hl("WildMenu", s:color_bg, s:color_primary, "")
+
+"+--- Search ---+
+call s:hl("IncSearch", s:color_bg_alt, s:color_bg_accent, "NONE")
+call s:hl("Search", s:color_bg_alt, s:color_bg_accent, "NONE")
+
+"+--- Tabs ---+
+call s:hl("TabLine", "NONE", s:color_bg_alt, "NONE")
+call s:hl("TabLineFill", "NONE", s:color_bg_accent, "NONE")
+call s:hl("TabLineSel", s:color_bg_alt, s:color_bg_accent, "NONE")
+
+" Window
+call s:hl("Title", "NONE", s:color_bg_alt, "NONE")
+call s:hl("VertSplit", s:color_secondary, "NONE", "NONE")
+
+" Syntax
+"
+call s:hl("Comment", s:color_syntax_d, "", "")
+
+call s:hl("Boolean", s:color_syntax_c, "", "")
+call s:hl("Character", s:color_syntax_b, "", "")
+call s:hl("Float", s:color_syntax_c, "", "")
+call s:hl("Number", s:color_syntax_c, "", "")
+call s:hl("String", s:color_syntax_a, "", "")
+call s:hl("Label", s:color_syntax_a, "", "")
+call s:hl("PreProc", s:color_syntax_b, "", "")
+call s:hl("Type", s:color_syntax_c, "", "")
+call s:hl("Todo", s:color_bg_alt, s:color_warn, "")
+
+call s:hl("Identifier", s:color_syntax_b, "", "bold")
+call s:hl("Conditional", s:color_syntax_b, "", "")
+call s:hl("Delimiter", s:color_syntax_b, "", "")
+call s:hl("Function", s:color_syntax_a, "", "")
+call s:hl("Keyword", s:color_syntax_b, "", "")
+call s:hl("Operator", s:color_syntax_a, "", "")
+call s:hl("Special", s:color_syntax_b, "", "")
+call s:hl("SpecialComment", s:color_syntax_c, "", "")
+call s:hl("Statement", s:color_syntax_b, "", "")
+
+call s:hl("Constant", s:color_syntax_a, "", "")
+call s:hl("Define", s:color_syntax_a, "", "")
+call s:hl("Exception", s:color_syntax_a, "", "")
+call s:hl("Include", s:color_syntax_a, "", "")
+call s:hl("Repeat", s:color_syntax_b, "", "")
+call s:hl("SpecialChar", s:color_syntax_b, "", "")
+call s:hl("StorageClass", s:color_syntax_b, "", "")
+call s:hl("Structure", s:color_syntax_a, "", "")
+call s:hl("Tag", s:color_syntax_a, "", "")
+call s:hl("Typedef", s:color_syntax_b, "", "")
+
+hi! link Macro Define
+hi! link PreCondit PreProc
+
+" CSS 
+call s:hl("cssAttributeSelector", s:color_syntax_a, "", "")
+call s:hl("cssDefinition", s:color_syntax_a, "", "bold")
+call s:hl("cssStringQ", s:color_syntax_a, "", "")
+call s:hl("cssBraces", s:color_syntax_a, "", "")
+call s:hl("cssProp", s:color_syntax_b, "", "")
+call s:hl("cssAttr", s:color_syntax_a, "", "bold")
+hi! link cssIdentifier cssDefinition
+hi! link cssClassName cssDefinition
+hi! link cssColor Number
+hi! link cssPseudoClass cssDefinition
+hi! link cssPseudoClassId cssPseudoClass
+hi! link cssVendor Keyword
+
+
+
+
+
+
+
+
+
+
+
+
 let s:nord0_gui = "#2E3440"
 let s:nord1_gui = "#3B4252"
 let s:nord2_gui = "#434C5E"
@@ -48,6 +223,9 @@ let s:nord13_term = "3"
 let s:nord14_term = "2"
 let s:nord15_term = "5"
 
+if !exists('g:nord_italic_comments')
+  let g:nord_italic_comments = 0
+endif
 
 function! s:hi(group, guifg, guibg, ctermfg, ctermbg, attr, guisp)
   let l:attr = a:attr
@@ -78,104 +256,8 @@ endfunction
 "+---------------+
 "+ UI Components +
 "+---------------+
-"+--- Attributes ---+
-call s:hi("Bold", "", "", "", "", "bold", "")
-call s:hi("Italic", "", "", "", "", "italic", "")
-call s:hi("Underline", "", "", "", "", "underline", "")
-
-"+--- Editor ---+
-call s:hi("ColorColumn", "", s:nord1_gui, "NONE", s:nord1_term, "", "")
-call s:hi("Cursor", s:nord0_gui, s:nord4_gui, "", "NONE", "", "")
-call s:hi("CursorLine", "", s:nord1_gui, "NONE", s:nord1_term, "NONE", "")
-call s:hi("iCursor", s:nord0_gui, s:nord4_gui, "", "NONE", "", "")
-call s:hi("LineNr", s:nord3_gui, s:nord0_gui, s:nord3_term, "NONE", "", "")
-call s:hi("MatchParen", s:nord0_gui, s:nord8_gui, s:nord1_term, s:nord8_term, "", "")
-call s:hi("NonText", s:nord2_gui, "", s:nord3_term, "", "", "")
-call s:hi("Normal", s:nord4_gui, s:nord0_gui, "NONE", "NONE", "", "")
-call s:hi("PMenu", s:nord4_gui, s:nord2_gui, "NONE", s:nord1_term, "NONE", "")
-call s:hi("PmenuSbar", s:nord4_gui, s:nord2_gui, "NONE", s:nord1_term, "", "")
-call s:hi("PMenuSel", s:nord8_gui, s:nord3_gui, s:nord8_term, s:nord3_term, "", "")
-call s:hi("PmenuThumb", s:nord8_gui, s:nord3_gui, "NONE", s:nord3_term, "", "")
-call s:hi("SpecialKey", s:nord3_gui, "", s:nord3_term, "", "", "")
-call s:hi("SpellBad", "", s:nord0_gui, "", "NONE", "undercurl", s:nord11_gui)
-call s:hi("SpellCap", "", s:nord0_gui, "", "NONE", "undercurl", s:nord13_gui)
-call s:hi("SpellLocal", "", s:nord0_gui, "", "NONE", "undercurl", s:nord5_gui)
-call s:hi("SpellRare", "", s:nord0_gui, "", "NONE", "undercurl", s:nord6_gui)
-call s:hi("Visual", "", s:nord2_gui, "", s:nord1_term, "", "")
-call s:hi("VisualNOS", "", s:nord2_gui, "", s:nord1_term, "", "")
-"+- Neovim Support -+
-call s:hi("healthError", s:nord11_gui, s:nord1_gui, s:nord11_term, s:nord1_term, "", "")
-call s:hi("healthSuccess", s:nord14_gui, s:nord1_gui, s:nord14_term, s:nord1_term, "", "")
-call s:hi("healthWarning", s:nord13_gui, s:nord1_gui, s:nord13_term, s:nord1_term, "", "")
-
-"+--- Gutter ---+
-call s:hi("CursorColumn", "", s:nord1_gui, "NONE", s:nord1_term, "", "")
-call s:hi("CursorLineNr", s:nord3_gui, s:nord0_gui, "NONE", "", "", "")
-call s:hi("Folded", s:nord3_gui, s:nord1_gui, s:nord3_term, s:nord1_term, "bold", "")
-call s:hi("FoldColumn", s:nord3_gui, s:nord0_gui, s:nord3_term, "NONE", "", "")
-call s:hi("SignColumn", s:nord1_gui, s:nord0_gui, s:nord1_term, "NONE", "", "")
-
-"+--- Navigation ---+
-call s:hi("Directory", s:nord8_gui, "", s:nord8_term, "NONE", "", "")
-
-"+--- Prompt/Status ---+
-call s:hi("EndOfBuffer", s:nord1_gui, "", s:nord1_term, "NONE", "", "")
-call s:hi("ErrorMsg", s:nord4_gui, s:nord11_gui, "NONE", s:nord11_term, "", "")
-call s:hi("ModeMsg", s:nord4_gui, "", "", "", "", "")
-call s:hi("MoreMsg", s:nord4_gui, "", "", "", "", "")
-call s:hi("Question", s:nord4_gui, "", "NONE", "", "", "")
-call s:hi("StatusLine", s:nord4_gui, s:nord0_gui, s:nord1_term, s:nord15_term, "NONE", "")
-call s:hi("StatusLineNC", s:nord4_gui, s:nord0_gui, s:nord10_term, "NONE", "NONE", "")
-call s:hi("WarningMsg", s:nord0_gui, s:nord13_gui, s:nord1_term, s:nord13_term, "", "")
-call s:hi("WildMenu", s:nord8_gui, s:nord3_gui, s:nord8_term, s:nord3_term, "", "")
 
-"+--- Search ---+
-call s:hi("IncSearch", s:nord1_gui, s:nord8_gui, s:nord1_term, s:nord8_term, "underline", "")
-call s:hi("Search", s:nord1_gui, s:nord8_gui, s:nord1_term, s:nord15_term, "NONE", "")
 
-"+--- Tabs ---+
-call s:hi("TabLine", s:nord4_gui, s:nord1_gui, "NONE", s:nord1_term, "NONE", "")
-call s:hi("TabLineFill", s:nord4_gui, s:nord1_gui, "NONE", s:nord1_term, "NONE", "")
-call s:hi("TabLineSel", s:nord8_gui, s:nord3_gui, s:nord8_term, s:nord3_term, "NONE", "")
-
-"+--- Window ---+
-call s:hi("Title", s:nord4_gui, "", "NONE", "", "NONE", "")
-call s:hi("VertSplit", s:nord2_gui, s:nord1_gui, s:nord3_term, s:nord1_term, "NONE", "")
-
-"+----------------------+
-"+ Language Base Groups +
-"+----------------------+
-call s:hi("Boolean", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Character", s:nord14_gui, "", s:nord14_term, "", "", "")
-call s:hi("Comment", s:nord3_gui, "", s:nord3_term, "", "italic", "")
-call s:hi("Conditional", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Constant", s:nord4_gui, "", "NONE", "", "", "")
-call s:hi("Define", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Delimiter", s:nord6_gui, "", s:nord6_term, "", "", "")
-call s:hi("Exception", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Float", s:nord15_gui, "", s:nord15_term, "", "", "")
-call s:hi("Function", s:nord8_gui, "", s:nord8_term, "", "", "")
-call s:hi("Identifier", s:nord4_gui, "", "NONE", "", "NONE", "")
-call s:hi("Include", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Keyword", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Label", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Number", s:nord15_gui, "", s:nord15_term, "", "", "")
-call s:hi("Operator", s:nord9_gui, "", s:nord9_term, "", "NONE", "")
-call s:hi("PreProc", s:nord9_gui, "", s:nord9_term, "", "NONE", "")
-call s:hi("Repeat", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Special", s:nord4_gui, "", "NONE", "", "", "")
-call s:hi("SpecialChar", s:nord13_gui, "", s:nord13_term, "", "", "")
-call s:hi("SpecialComment", s:nord8_gui, "", s:nord8_term, "", "italic", "")
-call s:hi("Statement", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("StorageClass", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("String", s:nord14_gui, "", s:nord14_term, "", "", "")
-call s:hi("Structure", s:nord9_gui, "", s:nord9_term, "", "", "")
-call s:hi("Tag", s:nord4_gui, "", "", "", "", "")
-call s:hi("Todo", s:nord13_gui, "", s:nord13_term, "NONE", "", "")
-call s:hi("Type", s:nord9_gui, "", s:nord9_term, "", "NONE", "")
-call s:hi("Typedef", s:nord9_gui, "", s:nord9_term, "", "", "")
-hi! link Macro Define
-hi! link PreCondit PreProc
 
 "+-----------+
 "+ Languages +
@@ -205,18 +287,6 @@ hi! link csPreCondit PreCondit
 hi! link csType Type
 hi! link csXmlTag SpecialComment
 
-call s:hi("cssAttributeSelector", s:nord7_gui, "", s:nord7_term, "", "", "")
-call s:hi("cssDefinition", s:nord7_gui, "", s:nord7_term, "", "NONE", "")
-call s:hi("cssIdentifier", s:nord7_gui, "", s:nord7_term, "", "underline", "")
-call s:hi("cssStringQ", s:nord7_gui, "", s:nord7_term, "", "", "")
-hi! link cssAttr Keyword
-hi! link cssBraces Delimiter
-hi! link cssClassName cssDefinition
-hi! link cssColor Number
-hi! link cssProp cssDefinition
-hi! link cssPseudoClass cssDefinition
-hi! link cssPseudoClassId cssPseudoClass
-hi! link cssVendor Keyword
 
 call s:hi("dosiniHeader", s:nord8_gui, "", s:nord8_term, "", "", "")
 hi! link dosiniLabel Type
